ferrocene/tools/self-test/src/targets.rs

```diff
@@ -70,15 +70,11 @@ fn check_target(
         return Ok(CheckTargetOutcome::Missing);
     }
 
-    check_libraries(
-        target,
-        &target_dir,
-        if target.std {
-            &["core", "alloc", "std", "test", "proc_macro"]
-        } else {
-            &["core", "alloc"]
-        },
-    )?;
+    let expected_libs = match target.std {
+        true => ["core", "alloc", "std", "test", "proc_macro"].as_slice(),
+        false => ["core", "alloc"].as_slice(),
+    };
+    check_libraries(target, &target_dir, expected_libs)?;
```

Comment:
does this work
```suggestion
    let expected_libs = match target.std {
        true => ["core", "alloc", "std", "test", "proc_macro"],
        false => ["core", "alloc"],
    };
    check_libraries(target, &target_dir, &expected_libs)?;
```

Reply:
I _think_ this does not compile, because the type of `expected_libs` is unclear, because the types in the both arms are conflicting . Is it `[&str; 5]` or is it `[&str; 2]`?